### ISSUE TITLE
mixed_linear_model._smw_solver argument shapes not specified

### ISSUE THREAD

**Issue by jbrockmendel** [statsmodels/statsmodels#4455]

**Comment by josef-pkt** [comment]
what does that mean?

more information in the description of an issue, please

**Comment by jbrockmendel** [comment]
<b>TLDR</b> For the purposes of writing tests, how should `BI` and `di` arguments be generated?

OK I think I've sorted out the shapes issue, but the input `BI` is still up in the air, and the docstring for smw_solver gives two conflicting definitions:

```
    The inverse matrix of B is block diagonal.  The upper left block
    is BI and the lower right block is a diagonal matrix containing
    di.
[...]
    BI : square symmetric ndarray
        The inverse of `B`.
```

Looking at the docstring for smw_logdet, that description is consistent with the first description above being correct and the second one being truncated:

```
    BI : square symmetric ndarray
        The upper left block of B^-1.
```

**Comment by jbrockmendel** [comment]
@kshedden any insight here?  I'd like to put together tests of these functions to rule them out as possible causes in #4414.

**Comment by jbrockmendel** [comment]
Closed by #4594
